File: web/src/api/nginx_domain_cert.js
import service from '@/utils/request'

// @Tags NginxDomainCert
// @Summary 创建NginxDomainCert
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.NginxDomainCert true "创建NginxDomainCert"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /Cert/createNginxDomainCert [post]
export const createNginxDomainCert = (data) => {
     return service({
         url: "/Cert/createNginxDomainCert",
         method: 'post',
         data
     })
 }


// @Tags NginxDomainCert
// @Summary 删除NginxDomainCert
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.NginxDomainCert true "删除NginxDomainCert"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /Cert/deleteNginxDomainCert [delete]
 export const deleteNginxDomainCert = (data) => {
     return service({
         url: "/Cert/deleteNginxDomainCert",
         method: 'delete',
         data
     })
 }

// @Tags NginxDomainCert
// @Summary 删除NginxDomainCert
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除NginxDomainCert"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /Cert/deleteNginxDomainCert [delete]
 export const deleteNginxDomainCertByIds = (data) => {
     return service({
         url: "/Cert/deleteNginxDomainCertByIds",
         method: 'delete',
         data
     })
 }

// @Tags NginxDomainCert
// @Summary 更新NginxDomainCert
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.NginxDomainCert true "更新NginxDomainCert"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /Cert/updateNginxDomainCert [put]
 export const updateNginxDomainCert = (data) => {
     return service({
         url: "/Cert/updateNginxDomainCert",
         method: 'put',
         data
     })
 }


// @Tags NginxDomainCert
// @Summary 用id查询NginxDomainCert
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.NginxDomainCert true "用id查询NginxDomainCert"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /Cert/findNginxDomainCert [get]
 export const findNginxDomainCert = (params) => {
     return service({
         url: "/Cert/findNginxDomainCert",
         method: 'get',
         params
     })
 }


// @Tags NginxDomainCert
// @Summary 分页获取NginxDomainCert列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取NginxDomainCert列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /Cert/getNginxDomainCertList [get]
 export const getNginxDomainCertList = (params) => {
     return service({
         url: "/Cert/getNginxDomainCertList",
         method: 'get',
         params
     })
 }

File: server/router/nginx_domain_cert.go
package router

import (
	"nginx-web/api/v1"
	"nginx-web/middleware"
	"github.com/gin-gonic/gin"
)

func InitNginxDomainCertRouter(Router *gin.RouterGroup) {
	NginxDomainCertRouter := Router.Group("Cert").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		NginxDomainCertRouter.POST("createNginxDomainCert", v1.CreateNginxDomainCert)   // 新建NginxDomainCert
		NginxDomainCertRouter.DELETE("deleteNginxDomainCert", v1.DeleteNginxDomainCert) // 删除NginxDomainCert
		NginxDomainCertRouter.DELETE("deleteNginxDomainCertByIds", v1.DeleteNginxDomainCertByIds) // 批量删除NginxDomainCert
		NginxDomainCertRouter.PUT("updateNginxDomainCert", v1.UpdateNginxDomainCert)    // 更新NginxDomainCert
		NginxDomainCertRouter.GET("findNginxDomainCert", v1.FindNginxDomainCert)        // 根据ID获取NginxDomainCert
		NginxDomainCertRouter.GET("getNginxDomainCertList", v1.GetNginxDomainCertList)  // 获取NginxDomainCert列表
	}
}
